mlir/lib/Conversion/ArithToEmitC/ArithToEmitCPass.cpp

```diff
@@ -0,0 +1,49 @@
+//===- ArithToEmitCPass.cpp - Arith to EmitC Pass ---------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements a pass to convert the Arith dialect to the EmitC
+// dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Conversion/ArithToEmitC/ArithToEmitCPass.h"
+
+#include "mlir/Conversion/ArithToEmitC/ArithToEmitC.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/EmitC/IR/EmitC.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+namespace mlir {
+#define GEN_PASS_DEF_CONVERTARITHTOEMITC
+#include "mlir/Conversion/Passes.h.inc"
+} // namespace mlir
+
+using namespace mlir;
+
+namespace {
+struct ConvertArithToEmitC
+    : public impl::ConvertArithToEmitCBase<ConvertArithToEmitC> {
+  void runOnOperation() override;
+};
+} // namespace
+
+void ConvertArithToEmitC::runOnOperation() {
+  ConversionTarget target(getContext());
+
+  target.addLegalDialect<emitc::EmitCDialect>();
+
```

Comment:
Do we need `target.addIllegalDialect<arith::ArithDialect>();` here? I think it would make this pass fail if we there are unhandled arith ops, which sounds better than failing at some later point (maybe inside the translation).

Reply:
With that change one could not use the *currently* implemented conversions **and** let the emitter emit code for `arith.constant` (which is still possible). Instead we could add `target.addIllegalOp<arith::AddFOp, arith::DivFOp, arith::MulFOp, arith::SubFOp>()` to mark ops for which we have conversions illegal.

Reply:
Then let's add `target.addIllegalDialect<arith::ArithDialect>();` with the PR that adds arith.constant conversion, right? (fyi @TinaAMD)

Reply:
After discussing different options offline with @simon-camp, I now marked the Arith dialect as illegal and `arith::ConstantOp` as legal. The latter can be dropped as part of the follow up PR that adds a conversion from `arith.constant` to EmitC. This follow up can then also remove the support for `arith.constant` from the emitter.